x-pack/plugins/fleet/common/constants/routes.ts

```diff
@@ -204,3 +204,9 @@ export const DOWNLOAD_SOURCE_API_ROUTES = {
   UPDATE_PATTERN: `${API_ROOT}/agent_download_sources/{sourceId}`,
   DELETE_PATTERN: `${API_ROOT}/agent_download_sources/{sourceId}`,
 };
+
+// API versioning constants
+export const OLDEST_PUBLIC_VERSION = '2023-10-31';
```

Comment:
I would call this CURRENT or LATEST_PUBLIC_VERSION.
Let's say we introduce a new version '2023-12-31' and want all Fleet usages to use it, we would have to replace the `OLDEST_PUBLIC_VERSION` everywhere, as the naming wouldn't be appropriate anymore.

Reply:
I can call it in a different way of course, but we are never going to replace all the current usages. The way it should work is "additive":
```
router.versioned.
    .post({
      path: '/api/my-app/foo/{id?}',
    })
    // oldest version, never going to change
    .addVersion(
      {
        version: OLDEST_PUBLIC_VERSION,
        validate: {
          request: endpointValidationSchema
          },
      },
      endpointHandler,
    )
    // second version released
    .addVersion(
      {
        version: '2023-12-31',
        validate: {
          request: newEndpointValidationSchema
          },
      },
      endpointHandler,
    )
```